sql/core/src/test/scala/org/apache/spark/sql/execution/metric/SQLMetricsSuite.scala

```diff
@@ -694,7 +694,8 @@ class SQLMetricsSuite extends SharedSparkSession with SQLMetricsTestUtils
   }
 
   test("SPARK-25497: LIMIT within whole stage codegen should not consume all the inputs") {
-    withSQLConf(SQLConf.WHOLESTAGE_CODEGEN_ENABLED.key -> "true") {
+    withSQLConf(SQLConf.WHOLESTAGE_CODEGEN_ENABLED.key -> "true",
+      SQLConf.REPLACE_HASH_WITH_SORT_AGG_ENABLED.key -> "false") {
```

Comment:
This unit test is testing multiple limit operators and hash aggregate operators in one single stage. Disable the rule here because sort aggregate does not support code-gen now, and will break the test logic.